libnetwork/cnmallocator/drivers_ipam.go

```diff
@@ -33,20 +32,13 @@ func initIPAMDrivers(r ipamapi.Registerer, netConfig *networkallocator.Config) e
 		str.WriteString(strconv.Itoa(int(netConfig.SubnetSize)))
 
 	}
-	if err := ipamutils.ConfigGlobalScopeDefaultNetworks(addressPool); err != nil {
-		return err
-	}
-	if addressPool != nil {
+
+	if len(addressPool) > 0 {
 		log.G(context.TODO()).Infof("Swarm initialized global default address pool to: " + str.String())
 	}
 
-	for _, fn := range [](func(ipamapi.Registerer) error){
-		builtinIpam.Register,
-		nullIpam.Register,
-	} {
-		if err := fn(r); err != nil {
-			return err
-		}
+	if err := ipams.Register(r, nil, []*ipamutils.NetworkToSplit(nil), addressPool); err != nil {
```

Comment:
What's with the typed-nil slice value?
```suggestion
	if err := ipams.Register(r, nil, nil, addressPool); err != nil {
```